lib/acts_as_ar_query.rb

```diff
@@ -194,6 +194,35 @@ def append_hash_arg(symbol, *val)
     end
   end
 
+  def append_hash_array_arg(symbol, default, *val)
+    val = val.flatten.compact
+    val = val.first if val.size == 1 && val.first.kind_of?(Hash)
```

Comment:
The user can pass in `nil, [], or {}` - I'll look into extract_options, but don't think we want that.